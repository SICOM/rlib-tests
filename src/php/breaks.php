<?php
$data[0] = array('company', 'region', 'district', 'store', 'sales');

$data[1] = array('Foo, Inc.', 'Region A','District X', 'Store #1', 1500.00);
$data[2] = array('Foo, Inc.', 'Region A','District X', 'Store #2', 1300.00);
$data[3] = array('Foo, Inc.', 'Region A','District Y', 'Store #3', 2000.00);
$data[4] = array('Foo, Inc.', 'Region A','District Y', 'Store #4', 1800.00);
$data[5] = array('Foo, Inc.', 'Region B','District M', 'Store #11', 2500.00);
$data[6] = array('Foo, Inc.', 'Region B','District M', 'Store #12', 2300.00);
$data[7] = array('Foo, Inc.', 'Region B','District N', 'Store #13', 3000.00);
$data[8] = array('Foo, Inc.', 'Region B','District N', 'Store #14', 2800.00);

$rlib =	rlib_init();
rlib_version();
rlib_add_datasource_array($rlib, "local_array");
rlib_add_query_as($rlib, "local_array", "data", "data");

$output_format = 'txt';

if (isset($argv[1]))
	$output_format = $argv[1];

rlib_add_report($rlib, "breaks.xml");
rlib_set_output_format_from_text($rlib, $output_format);
rlib_set_output_parameter($rlib, "debugging", "yes");
rlib_set_output_parameter($rlib, "only_quote_strings", "yes");
rlib_set_locale($rlib, "en_US");
rlib_execute($rlib);
// header(rlib_get_content_type($rlib));
rlib_spool($rlib);
rlib_free($rlib);
	
?>
