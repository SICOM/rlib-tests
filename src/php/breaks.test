COMMAND="$PHP breaks.php"
EXTRA_COMMAND=
FORMATS="pdf txt html"
EXPECTED_pdf="1 2 3"
EXPECTED_xml=
EXPECTED_txt=
EXPECTED_csv=
EXPECTED_html="1 2"
