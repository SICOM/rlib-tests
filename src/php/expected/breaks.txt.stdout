Breaks Example                          
Company District Region Store                                           Sales 
Foo, Inc.                              $1500.00
 Region A                               $1500.00
  District X                            $1500.00
   Store #1                            $1500.00 
   Store #2                            $1300.00 
  District X                            $2800.00
  District Y                            $2000.00
   Store #3                            $2000.00 
   Store #4                            $1800.00 
  District Y                            $3800.00
 Region A                               $6600.00
 Region B                               $2500.00
  District M                            $2500.00
   Store #11                           $2500.00 
   Store #12                           $2300.00 
  District M                            $4800.00
  District N                            $3000.00
   Store #13                           $3000.00 
   Store #14                           $2800.00 
  District N                            $5800.00
 Region B                              $10600.00
Foo, Inc.                              $17200.00
